### PR TITLE
Guard length of url for bz reporting

### PR DESCRIPTION
**Pull request by rvykydal**
Low priority, the first commit can be perhaps merged separately.
https://issues.redhat.com/browse/INSTALLER-3612

Example of truncated comment:
(The message should appear really exceptionally - we don't expect automatically generated Detail/comment to exceed the 7000 characters. It is there so it is clear what really happened (the comment was truncated) in such a rare case.)

Screenshot from 2023-08-22 14-34-38

### CODE REVIEW

File: ui/webui/src/components/Error.jsx [hunk 1]
```diff
@@ -53,6 +53,21 @@ export const bugzillaPrefiledReportURL = (productQueryData) => {
     return reportURL.href;
 };
 
+const ensureMaximumReportURLLength = (reportURL) => {
+    const newUrl = new URL(reportURL);
+    const searchParamsLimits = [
+        { param: "short_desc", length: 256, message: null },
+        { param: "comment", length: 7800, message: "\nTHE COMMENT EXCEEDING THE LENGTH LIMIT WAS TRUNCATED" },
```
Comment (KKoukiou) [inline ui/webui/src/components/Error.jsx:60]: Should not there be an error in the dialog form instead of filling this warning in the URL?

Comment (rvykydal) [reply ui/webui/src/components/Error.jsx:60]: The comment (pre-filed Detail) is generated by Anaconda, there is no user input at this point (currently it is used only in CriticalError where it is the exception content) so I don't think it makes sense to inform user about it. It is meant as an information for developers, not for the user. So maybe we should remove it completely (and just log it) so it does not confuse user in the BZ report form.

File: ui/webui/src/components/Error.jsx [hunk 2]
```diff
@@ -83,6 +98,7 @@ export const BZReportModal = ({
     }, []);
 
     const openBZIssue = (reportURL, logFile, logContent) => {
+        reportURL = ensureMaximumReportURLLength(reportURL);
         reportURL = addLogAttachmentCommentToReportURL(reportURL, logFile);
```
Comment (VladimirSlavik) [inline ui/webui/src/components/Error.jsx:102]: Is this the right order? First you trim it and then add the comment.

File: ui/webui/src/components/Error.jsx (hunk 1 above)
Comment (VladimirSlavik) [reply ui/webui/src/components/Error.jsx:62]: I just realized, there is no mention of what's the actual limit? What is 7800?

Comment (rvykydal) [reply ui/webui/src/components/Error.jsx:62]: > I just realized, there is no mention of what's the actual limit? What is 7800?

I'll add a comment in the code.

File: ui/webui/src/components/Error.jsx (hunk 2 above)
Comment (rvykydal) [reply ui/webui/src/components/Error.jsx:102]: Yes it is. There number is taking the added comment (of ~ static length) into account. We want the added comment to be there always, irrespective of the message being cut. I'll explain it in a comment in the code as well.

**Comment by rvykydal** [comment]
So I added some comments in the code and removed the message informing about the Detail being cut from the Detail. The message asking to attach the log is still added after the cut detail.

Screenshot from 2023-09-05 15-29-40

**Review by VladimirSlavik — APPROVED** [review]
Thanks!

**Pull request merged**